test/integration/backward_compatible/parallel/topic/TopicTest.js

```diff
@@ -0,0 +1,146 @@
+        await topic.publish('item-value');
+
+        await new Promise((resolve) => {
+            const interval = setInterval(() => {
+                if (collector.events.length === 1) {
+                    clearInterval(interval);
+                    const event = collector.events[0];
+                    expect(event.messageObject).to.equal('item-value');
+                    expect(event.publishingTime).to.exist;
+                    if (event.publishingTime) {
+                        expect(event.publishingTime.toNumber()).to.be.greaterThan(0);
+                    }
+                    resolve();
+                }
+            }, 100);
+        });
+    });
+
+    it('removes listener', async function() {
+        class Collector {
+            constructor() {
+                this.events = [];
+            }
+
+            onMessage(event) {
+                this.events.push(event);
+            }
+        }
+
+        const collector = new Collector();
+        const reg_id = await topic.addListener(collector.onMessage.bind(collector));
+        await topic.removeListener(reg_id);
+        await topic.publish('item-value');
+
+        setTimeout(() => {
+            expect(collector.events.length).to.be(0);
+        }, 5000);
+    });
+
+    it('tests publish', async function () {
+        let count = 0;
+        const receivedValues = [];
+
+        await topic.addListener((message) => {
+            count++;
+            receivedValues.push(message.messageObject);
+        });
+
+        const message = 'message';
+        await topic.publish(message);
+
+        await TestUtil.assertTrueEventually(async () => {
+            expect(count).to.equal(1);
+            expect(receivedValues).to.deep.equal([message]);
+        });
+    });
+
+    it('tests publishAll', async function() {
+        let count = 0;
+        const receivedValues = [];
+
+        await topic.addListener((message) => {
+            count++;
+            receivedValues.push(message.messageObject);
+        });
+
+        const messages = ['message 1', 'message 2', 'message 3'];
+        await topic.publishAll(messages);
+
+        await TestUtil.assertTrueEventually(async () => {
+            expect(count).to.equal(messages.length);
+            expect(receivedValues).to.have.members(messages);
+        });
+    });
+
+    it('tests publishAll with one null element in an array', async function() {
+        const messages = [1, null, 3];
+        await expect(() => topic.publishAll(messages)).to.throw('Non null value expected.');
+    });
+
+    it('tests publishAll with null array', async function() {
+        const messages = null;
+        await expect(() => topic.publishAll(messages)).to.throw('Non null value expected.');
+    });
+
+    it('tests publishAll with null elements array', async function() {
+        const messages = [null, null, null];
+        await expect(() => topic.publishAll(messages)).to.throw('Non null value expected.');
```

Comment:
You should not need this async/await in these tests.

Reply:
Right, fixed